web/src/components/product/ProductSelectionProgress.jsx

```diff
@@ -0,0 +1,162 @@
+    if (!message) return "";
+
+    return (current === 0) ? message : `${message} (${current}/${total})`;
+  };
+
+  const stepProperties = (progress) => {
+    const properties = {
+      variant: variant(progress),
+      isCurrent: isCurrent(progress),
+      description: description(progress)
+    };
+
+    if (properties.isCurrent) properties.icon = <Spinner />;
+
+    return properties;
+  };
+
+  // Emulates progress for product selection step.
+  const productProgress = () => {
+    if (!storageProgress.start) return { start: true, current: 1 };
+
+    return { start: true, current: 0 };
+  };
+
+  /** @todo Add aria-label to steps, describing its status and variant. */
+  return (
+    <ProgressStepper isCenterAligned aria-label="Preparing selected product, please wait">
+      <ProgressStep
+        id="product-step"
+        titleId="product-step-title"
+        {...stepProperties(productProgress())}
+      >
+        {selectedProduct.name}
+      </ProgressStep>
+      <ProgressStep
+        id="storage-step"
+        titleId="storage-step-title"
+        {...stepProperties(storageProgress)}
+      >
+        {_("Prepare disk")}
+      </ProgressStep>
+      <ProgressStep
+        id="software-step"
+        titleId="software-step-title"
+        {...stepProperties(softwareProgress)}
+      >
+        {_("Configure software")}
+      </ProgressStep>
+    </ProgressStepper>
+  );
+};
+
+/**
+ * @component
+ *
+ * Shows progress steps when a product is selected.
+ *
+ * @note Some details are hardcoded (e.g., the steps, the order, etc). The progress API has to be
+ *  improved.
+ */
+function ProductSelectionProgress() {
+  const { cancellablePromise } = useCancellablePromise();
+  const { storage, software } = useInstallerClient();
+  const { selectedProduct } = useProduct();
+  const [storageProgress, setStorageProgress] = useState({});
+  const [softwareProgress, setSoftwareProgress] = useState({});
+
+  useEffect(() => {
+    const updateProgress = (progress) => {
+      if (progress.current > 0) progress.start = true;
+      setStorageProgress(p => ({ ...p, ...progress }));
+    };
+
+    cancellablePromise(storage.getProgress()).then(updateProgress);
+
+    return storage.onProgressChange(updateProgress);
+  }, [cancellablePromise, setStorageProgress, storage]);
+
+  useEffect(() => {
+    const updateProgress = (progress) => {
+      if (progress.current > 0) progress.start = true;
+      setSoftwareProgress(p => ({ ...p, ...progress }));
+      // Let's assume storage was started too.
+      setStorageProgress(p => ({ ...p, start: progress.start }));
+    };
+
+    cancellablePromise(software.getProgress()).then(updateProgress);
+
+    return software.onProgressChange(updateProgress);
+  }, [cancellablePromise, setSoftwareProgress, software]);
+
+  return (
+    <Center style={{ blockSize: "100vh" }}>
+      <Grid hasGutter>
+        <GridItem sm={8} smOffset={2}>
+          <Card>
```

Comment:
I'd use isPlain here

```suggestion
          <Card isPlain>
```